autogpt/prompts/prompt.py

```diff
@@ -46,13 +46,304 @@ def build_default_prompt_generator() -> PromptGenerator:
     return prompt_generator
 
 
-def construct_main_ai_config() -> AIConfig:
+def construct_main_ai_config(config_file: str = None) -> AIConfig:
     """Construct the prompt for the AI to respond to
 
+    Args:
+        config_file (str, optional): The path to the AI configuration file. If not provided, the default SAVE_FILE will be used.
+
     Returns:
-        str: The prompt string
+        AIConfig: The chosen or created AIConfig instance
     """
-    config = AIConfig.load(CFG.ai_settings_file)
+    if config_file is None:
+        config_file = AIConfig.SAVE_FILE
+
+    should_save_config = True
+
+    while True:
+        logger.typewriter_log(
+            f"Attempting to load configuration from: {config_file}",
+            Fore.GREEN,
+        )
+        all_configs = AIConfig.load_all(config_file)
+
+        if config_file in all_configs:
+            config = all_configs[config_file]
+            break
+        elif len(all_configs) > 0:
+            logger.typewriter_log(
+                "Welcome back! ",
+                Fore.GREEN,
+                "Configuration(s) detected. Please select one:",
+                speak_text=True,
+            )
+            for i, cfg_name in enumerate(all_configs.keys(), start=1):
+                logger.typewriter_log(f"{i}. {cfg_name}")
+            logger.typewriter_log(f"{len(all_configs) + 1}. Create new configuration")
+            logger.typewriter_log(
+                f"{len(all_configs) + 2}. Change existing configuration"
+            )
+            logger.typewriter_log(
+                f"{len(all_configs) + 3}. Delete existing configuration"
+            )
+            selection = clean_input("Please choose a number: ")
+            if selection.isdigit() and 0 < int(selection) <= len(all_configs) + 3:
```

Comment:
It doesn't feel great to be so reliant on list indexing / `len`. I think adding an intermediate set of variables or data structure indicating the user's selection would be really nice. That way the logic below reads cleaner, and if we decide to change the way that the menu works we wouldn't need to change all of the code.

Reply:
True it should also be DRY and easier to maintain, will look at it